src/notification-banner/notification-banner.tsx

```diff
@@ -40,35 +47,91 @@ export const NBComponent = ({
     // =============================================================================
     // EVENT HANDLERS
     // =============================================================================
-    const handleDismiss = () => {
+    const handleDismiss = (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation();
         setVisible(false);
 
         if (dismissible && onDismiss) onDismiss();
     };
 
+    const handleBannerClick = (event: React.MouseEvent) => {
+        if (!onClick) return;
+        event.stopPropagation();
+        onClick();
+    };
+
+    const handleActionButtonOnClick = (event: React.MouseEvent) => {
+        event.stopPropagation();
+        if (!actionButton.onClick) {
+            handleBannerClick(event);
+            return;
+        }
+        actionButton.onClick();
+    };
+
     // =============================================================================
     // RENDER FUNCTIONS
     // =============================================================================
     if (!isVisible) return null;
 
+    const renderDismiss = () => (
+        <StyledIconButton
+            onClick={handleDismiss}
+            id={formatId("dismiss-button", id)}
+            data-testid={formatId("dismiss-button", testId)}
+            focusHighlight={false}
+        >
+            <StyledIcon />
+        </StyledIconButton>
+    );
+
+    const renderActionButton = () => (
+        <ActionButton
+            id={formatId("action-button", id)}
+            data-testid={formatId("action-button", testId)}
+            onClick={handleActionButtonOnClick}
+            {...actionButton}
+        >
+            {actionButton.children}
+        </ActionButton>
+    );
+
+    const renderChildren = () => {
+        if (
+            !maxCollapsedHeight ||
+            !contentRef.current ||
+            (contentHeight && contentHeight < maxCollapsedHeight)
+        ) {
+            return children;
+        } else {
+            return (
+                <ContentWrapper $maxCollapsedHeight={maxCollapsedHeight}>
+                    {children}
+                </ContentWrapper>
+            );
+        }
+    };
+
     return (
-        <Wrapper ref={forwardedRef} $sticky={sticky} {...otherProps}>
+        <Wrapper
+            role={handleBannerClick ? "button" : "banner"}
```

Comment:
in addition to role `button`, focus and key presses also need to be handled https://www.w3.org/WAI/ARIA/apg/patterns/button/examples/button/#kbd_label

hence it might be easier to render an invisible button instead